src/nvme/types.h

```diff
@@ -9007,4 +9025,324 @@ struct nvme_ns_mgmt_host_sw_specified {
 };
 #endif /* SWIG */
 
+/**
+ * enum nvme_lm_select - Select (SEL): This field specifies the type of management operation to
+ * perform.
+ * @NVME_LM_SELECT_SHIFT:			Shift to set Select (SEL) field in all live
+ *						migration commands.
+ * @NVME_LM_SELECT_MASK:			Mask to set SEL field.
+ * @NVME_LM_CREATE_CONTROLLER_DATA_QUEUE:	Controller Data Queue - Create
+ * @NVME_LM_DELETE_CONTROLLER_DATA_QUEUE:	Controller Data Queue - Delete
+ * @NVME_LM_LOG_USER_DATA_CHANGES:		Track Send - Log User Data Changes
+ * @NVME_LM_TRACK_MEMORY_CHANGES:		Track Send - Track Memory Changes
+ * @NVME_LM_SUSPEND:				Migration Send - Suspend
+ * @NVME_LM_RESUME:				Migration Send - Resume
+ * @NVME_LM_SET_CONTROLLER_STATE:		Migration Send - Set Controller State
+ * @NVME_LM_GET_CONTROLLER_STATE:		Migration Receive - Get Controller State
+ */
+enum nvme_lm_select {
+	NVME_LM_SELECT_SHIFT			= 0,
+	NVME_LM_SELECT_MASK			= 0xff,
+	/* Controller Data Queue */
+	NVME_LM_CREATE_CONTROLLER_DATA_QUEUE	= 0,
+	NVME_LM_DELETE_CONTROLLER_DATA_QUEUE	= 1,
+	/* Track Send */
+	NVME_LM_LOG_USER_DATA_CHANGES		= 0,
+	NVME_LM_TRACK_MEMORY_CHANGES		= 1,
+	/* Migration Send */
+	NVME_LM_SUSPEND				= 0,
+	NVME_LM_RESUME				= 1,
+	NVME_LM_SET_CONTROLLER_STATE		= 2,
+	/* Migration Receive */
+	NVME_LM_GET_CONTROLLER_STATE		= 0,
```

Comment:
I fiend this very surprising that the `select` field from  different commands are mixed into one enum. Do we have this pattern already somewhere else?

Reply:
All the LM commands share the same cdw10 structure. I thought it better organized in one select block.

Would you rather I have an enum per command, where all fields pertaining to that command are located in one block? Or an enum per command per field?

Reply:
While I understand the motivation to interpret the spec and make it a bit more usable I am not sure if this is a good thing in the long run. We didn't do this for the rest of the spec. Maybe @keithbusch has some input on this.